pkg/skaffold/deploy/kubectl/matcher/kind.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package matcher
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	kindKey = "kind"
+)
+
+// Kind matches the value of "Kind" key with values not listed in the forbiddenValues.
+type Kind struct {
+	forbiddenValues map[string]struct{}
+}
+
+func (k *Kind) Matches(v interface{}) bool {
```

Comment:
```suggestion
func (k *Kind) MatchesValue(v interface{}) bool {
```

Reply:
This is a interface defines in https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/deploy/kubectl/matcher.go#L22
```
// Matcher is used by Replacer to match object blob to replace based
// on a manifest key in the Manifest
// Note: If the manifest key is not present, the replacer will replace.
type Matcher interface {
	MatchesKey(key string) bool
	Matches(v interface{}) bool
}
```

Reply:
If we have to do this, we need to write a `Ignorer` interface.
WDYT?